src/main/java/persistence/sql/dml/DMLQueryBuilder.java

```diff
@@ -0,0 +1,51 @@
+package persistence.sql.dml;
+
+import jakarta.persistence.*;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+public abstract class DMLQueryBuilder {
+
+    String getTableName(Class<?> field) {
+        if (field.isAnnotationPresent(Table.class)) {
+            Table table = field.getAnnotation(Table.class);
+            return !table.name().isEmpty() ? table.name() : field.getSimpleName();
+        }
+        return field.getSimpleName();
+    }
+
+    String columnsClause(Class<?> field) {
+        return Arrays.stream(field.getDeclaredFields())
+                .filter(c -> !c.isAnnotationPresent(Id.class) || !c.isAnnotationPresent(GeneratedValue.class))
+                .filter(c -> !c.isAnnotationPresent(Transient.class)).map(this::getColumnName).collect(Collectors.joining(", "));
+    }
+
+    String valueClause(Object entity) {
+        return Arrays.stream(entity.getClass().getDeclaredFields())
+                .filter(field -> !field.isAnnotationPresent(Id.class) || !field.isAnnotationPresent(GeneratedValue.class))
+                .filter(field -> !field.isAnnotationPresent(Transient.class)).map(field -> {
+                    field.setAccessible(true);
+                    try {
+                        Object value = field.get(entity);
+                        if (value instanceof String) {
+                            return "'" + value + "'";
+                        }
+                        return value != null ? value.toString() : "NULL";
+                    } catch (IllegalAccessException e) {
+                        throw new RuntimeException(e);
+                    }
+                }).collect(Collectors.joining(", "));
```

Comment:
```suggestion
            public String valueClause(Object entity) {
        return Arrays.stream(entity.getClass().getDeclaredFields())
                .filter(this::isIncludedField)
                .map(field -> formatFieldValue(field, entity))
                .collect(Collectors.joining(", "));
    }
```

책임에 맞는 메소드 분리와 책임들이 모여 객체를 이루고 객체가 모여 팩키지가 된다고 봤을때,
저는 우선 작은범위에 해당하는 함수 내에서 복잡해 보이는 로직들안에서 부터 리팩토링을 진행볼것 같아요.

로직이 의미하는 적절한 이름을 도출해내고 메소드로 분리해보는것이죠.